metron-streaming/Metron-Solr/src/main/java/org/apache/metron/writer/solr/SolrWriter.java

```diff
@@ -65,7 +71,9 @@ public void write(String sourceType, Configurations configurations, List<Tuple>
       }
       UpdateResponse response = solr.add(document);
     }
-    solr.commit(getCollection(configurations));
+    if (shouldCommit) {
+      solr.commit(getCollection(configurations));
+    }
   }
```

Comment:
I think in solrconfig.xml the autoSoftCommit just needs to be set to something:

https://github.com/apache/incubator-metron/pull/67/files#diff-1a81ac5587a67fc61fda38750c61ffa3R195

If you can tolerate some amount of delay from when new documents are added to when they are visible in a search, then maybe 60 seconds would be a good starting point?